huaweicloud/services/cdm/resource_huaweicloud_cdm_link.go

```diff
@@ -0,0 +1,310 @@
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/logp"
+)
+
+const (
+	configPref = "linkConfig."
+)
+
+func ResourceCdmLink() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceCdmLinkCreate,
+		ReadContext:   resourceCdmLinkRead,
+		UpdateContext: resourceCdmLinkUpdate,
+		DeleteContext: resourceCdmLinkDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"cluster_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"connector": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"config": {
+				Type:     schema.TypeMap,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Required: true,
+			},
+
+			"password": {
+				Type:      schema.TypeString,
+				Optional:  true,
+				Sensitive: true,
+			},
+
+			"access_key": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				RequiredWith: []string{"security_key"},
+			},
+
+			"security_key": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Sensitive:    true,
+				RequiredWith: []string{"access_key"},
+			},
+
+			"enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+		},
+	}
+}
+
+func resourceCdmLinkCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+	client, err := config.CdmV11Client(region)
+	if err != nil {
+		return diag.Errorf("error creating CDM v1 client, err=%s", err)
+	}
+
+	opts := link.LinkCreateOpts{
+		Links: []link.Link{
+			{
+				Name:             d.Get("name").(string),
+				ConnectorName:    d.Get("connector").(string),
+				Enabled:          utils.Bool(d.Get("enabled").(bool)),
+				LinkConfigValues: buildLinkConfigParamter(d),
+			},
+		},
+	}
+
+	clusterId := d.Get("cluster_id").(string)
+	logp.Printf("[DEBUG] Creating CDM link opts: %#v", opts)
+
+	rst, createErr := link.Create(client, clusterId, opts)
+	if createErr != nil {
+		return fmtp.DiagErrorf("Error creating CDM link: %s", createErr)
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s", clusterId, rst.Name))
+
+	return resourceCdmLinkRead(ctx, d, meta)
+}
+
+func buildLinkConfigParamter(d *schema.ResourceData) link.LinkConfigs {
+	var configs []link.Input
+	configRaw := d.Get("config").(map[string]interface{})
+	connector := d.Get("connector").(string)
+	if len(configRaw) > 0 {
```

Comment:
If the length of configRaw is 0, suggest return an error directly, and do not continue to call the API.